### PR TITLE
[CARBONDATA-3662]: Changes to show metacache command

### PR DESCRIPTION
**Pull request by vikramahuja1001**
### Why is this PR needed?
1. There is no command to show the cache occupied by each of the executors(only applicable for index server). 
This command will not show per table, only per node
Example:
Driver - X bytes
Executor1 - Y bytes
Executor2 - Z bytes
  
2. Show metacache command does not show the complete cache used used by IndexServer side and Driver side. It just shows the total of both of them.
 
 ### What changes were proposed in this PR?
 1. To change *show metacache* command to *show executor metacache*, where the "executor" keyword is optional.
2.  Added TOTAL cache value separately for Index Server and Driver.
    
 ### Does this PR introduce any user interface change?
 - No
 - Yes. (please explain the change and update document)

 ### Is any new testcase added?
 - No
 - Yes

### CODE REVIEW

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/1524/

**Comment by vikramahuja1001** [comment]
retest this please

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/1567/

**Comment by vikramahuja1001** [comment]
@kunal642 , @akashrn5 , please review

File: integration/spark2/src/main/scala/org/apache/carbondata/indexserver/DistributedShowCacheRDD.scala [hunk 1]
```diff
@@ -71,10 +75,21 @@ class DistributedShowCacheRDD(@transient private val ss: SparkSession, tableUniq
                 .getTableUniqueName
             } else {
               dataMap.getDataMapSchema.getRelationIdentifier.getDatabaseName + "_" + dataMap
-              .getDataMapSchema.getDataMapName
+                .getDataMapSchema.getDataMapName
+            }
+            if (executorCache == true) {
```
Comment (kunal642) [inline integration/spark2/src/main/scala/org/apache/carbondata/indexserver/DistributedShowCacheRDD.scala]: "== true " is not required

File: integration/spark2/src/main/scala/org/apache/carbondata/indexserver/IndexServer.scala [hunk 2]
```diff
@@ -57,7 +57,7 @@ trait ServerInterface {
   /**
    * Get the cache size for the specified tables.
    */
-  def showCache(tableIds: String) : Array[String]
+  def showCache(executorCache: Boolean, tableIds: String) : Array[String]
```
Comment (kunal642) [inline integration/spark2/src/main/scala/org/apache/carbondata/indexserver/IndexServer.scala]: Add to the end

File: integration/spark2/src/main/scala/org/apache/carbondata/indexserver/IndexServer.scala [hunk 3]
```diff
@@ -205,14 +205,18 @@ object IndexServer extends ServerInterface {
     }
   }
 
-  override def showCache(tableId: String = ""): Array[String] = doAs {
+  override def showCache(executorCache: Boolean, tableId: String = ""): Array[String] = doAs {
     val jobgroup: String = "Show Cache " + (tableId match {
-      case "" => "for all tables"
+      case "" =>
+        executorCache match {
```
Comment (kunal642) [inline integration/spark2/src/main/scala/org/apache/carbondata/indexserver/IndexServer.scala]: replace with if else

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala [hunk 4]
```diff
@@ -74,7 +92,15 @@ case class CarbonShowCacheCommand(tableIdentifier: Option[TableIdentifier],
       /**
        * Assemble result for database
        */
-      getAllTablesCache(sparkSession)
+      if (showExecutorCache == false) {
```
Comment (kunal642) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala]: "==false " not required...Change in all other places also

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala [hunk 5]
```diff
@@ -41,25 +42,42 @@ import org.apache.carbondata.spark.util.CarbonScalaUtil
...
+        Seq(
+          AttributeReference("Database and Table", StringType, nullable = false)(),
+          AttributeReference("Index size", StringType, nullable = false)(),
+          AttributeReference("Datamap size", StringType, nullable = false)(),
+          AttributeReference("Dictionary size", StringType, nullable = false)(),
+          AttributeReference("Cache Location", StringType, nullable = false)())
+      } else {
+        if (!isDistributedPruningEnabled) {
+          Seq(
+            AttributeReference("Database and Table", StringType, nullable = false)(),
```
Comment (kunal642) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala]: Change "Database and Table" to "Identifier"

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala [hunk 6]
```diff
@@ -41,25 +42,42 @@ import org.apache.carbondata.spark.util.CarbonScalaUtil
...
-      Seq(
-        AttributeReference("Database", StringType, nullable = false)(),
-        AttributeReference("Table", StringType, nullable = false)(),
-        AttributeReference("Index size", StringType, nullable = false)(),
-        AttributeReference("Datamap size", StringType, nullable = false)(),
-        AttributeReference("Dictionary size", StringType, nullable = false)(),
-        AttributeReference("Cache Location", StringType, nullable = false)())
+      val isDistributedPruningEnabled = CarbonProperties.getInstance()
+        .isDistributedPruningEnabled("", "")
+      if (showExecutorCache == false) {
```
Comment (kunal642) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala]: 1. simplyfy the if else block
2. Block in case of indexserver=false

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala [hunk 7]
```diff
@@ -110,13 +137,42 @@ case class CarbonShowCacheCommand(tableIdentifier: Option[TableIdentifier],
     }
   }
 
+  def getAllExecutorCache(sparkSession: SparkSession): Seq[Row] = {
+    val isDistributedPruningEnabled = CarbonProperties.getInstance()
+      .isDistributedPruningEnabled("", "")
+    if (!isDistributedPruningEnabled) {
+      getAllTablesCache(sparkSession)
+    }
+    else {
```
Comment (kunal642) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala]: fix the indentation for the whole PR

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala [hunk 8]
```diff
@@ -110,13 +137,42 @@ case class CarbonShowCacheCommand(tableIdentifier: Option[TableIdentifier],
...
     val cache = CacheProvider.getInstance().getCarbonCache
     val isDistributedPruningEnabled = CarbonProperties.getInstance()
       .isDistributedPruningEnabled("", "")
-    if (cache == null && !isDistributedPruningEnabled) {
-      return makeEmptyCacheRows(currentDatabase)
+    if (!isDistributedPruningEnabled) {
+      if (cache == null) {
+        return makeEmptyCacheRows(currentDatabase, "DRIVER")
+      }
+      if (cache.getCurrentSize == 0) {
```
Comment (kunal642) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala]: make as  if (cache == null || cache.getCurrentSize == 0) {

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala [hunk 9]
```diff
@@ -127,6 +183,8 @@ case class CarbonShowCacheCommand(tableIdentifier: Option[TableIdentifier],
             carbonTables += carbonTable
           }
         } catch {
+          case ex: AnalysisException =>
+            LOGGER.info("Unable to access Carbon table object for table" + tableIdent.table)
```
Comment (kunal642) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala]: make this log as debug

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala [hunk 10]
```diff
@@ -171,39 +252,33 @@ case class CarbonShowCacheCommand(tableIdentifier: Option[TableIdentifier],
...
-          indexdbIndexSize,
-          indexdbDatamapSize,
-          driverdbDictSize,
-          "INDEX SERVER")
-      ) ++ indexServerRows).collect {
-        case row if row.getLong(2) != 0L || row.getLong(3) != 0L || row.getLong(4) != 0L =>
-          Row(row.get(0), row.get(1), bytesToDisplaySize(row.getLong(2)),
-            bytesToDisplaySize(row.getLong(3)), bytesToDisplaySize(row.getLong(4)), "INDEX SERVER")
+    val indexDisplayRows = if (isDistributedPruningEnabled) {
+      if (indexServerRows.nonEmpty) {
```
Comment (kunal642) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala]: revert unnecessary changes

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala [hunk 11]
```diff
@@ -171,39 +252,33 @@ case class CarbonShowCacheCommand(tableIdentifier: Option[TableIdentifier],
...
+        ) ++ indexServerRows).collect {
+          case row if row.getLong(1) != 0L || row.getLong(2) != 0L || row.getLong(3) != 0L =>
+            Row(row.get(0),
+              bytesToDisplaySize(row.getLong(1)),
+              bytesToDisplaySize(row.getLong(2)),
+              bytesToDisplaySize(row.getLong(3)),
+              "INDEX SERVER")
+        }
+      } else {
+        makeEmptyCacheRows(currentDatabase, "INDEXSERVER")
```
Comment (kunal642) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala]: If there is nothing in the cache then no need to display anything.
We can remove this method itself and simplyfy the conditions.

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala [hunk 12]
```diff
@@ -135,23 +193,46 @@ case class CarbonShowCacheCommand(tableIdentifier: Option[TableIdentifier],
...
       carbonTables.flatMap {
         carbonTable =>
           try {
-            makeRows(getTableCacheFromDriver(sparkSession, carbonTable), carbonTable)
+            val row = makeRows(getTableCacheFromDriver(sparkSession, carbonTable), carbonTable)
+            var res: List[Any] = null
+            for (i <- row.toList) {
+              res = i.toSeq.toList
+            }
+            if (res(1) == 0 && res(2) == 0 && res(3) == 0) {
```
Comment (kunal642) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala]: can be done in makeRows method

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala [hunk 13]
```diff
@@ -135,23 +193,46 @@ case class CarbonShowCacheCommand(tableIdentifier: Option[TableIdentifier],
...
 
     val driverRows = if (cache != null) {
       carbonTables.flatMap {
         carbonTable =>
           try {
-            makeRows(getTableCacheFromDriver(sparkSession, carbonTable), carbonTable)
+            val row = makeRows(getTableCacheFromDriver(sparkSession, carbonTable), carbonTable)
+            var res: List[Any] = null
+            for (i <- row.toList) {
+              res = i.toSeq.toList
```
Comment (kunal642) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala]: No need to convert to List. same can be done for Seq also

File: integration/spark2/src/main/scala/org/apache/spark/sql/parser/CarbonSpark2SqlParser.scala [hunk 14]
```diff
@@ -562,7 +563,13 @@ class CarbonSpark2SqlParser extends CarbonDDLSqlParser {
   protected lazy val showCache: Parser[LogicalPlan] =
     SHOW ~> METACACHE ~> opt(ontable) <~ opt(";") ^^ {
       case table =>
-        CarbonShowCacheCommand(table)
+        CarbonShowCacheCommand(false, table)
+    }
+
+  protected lazy val showExecutorCache: Parser[LogicalPlan] =
+    SHOW ~> EXECUTOR ~> METACACHE ~> opt(ontable) <~ opt(";") ^^ {
```
Comment (kunal642) [inline integration/spark2/src/main/scala/org/apache/spark/sql/parser/CarbonSpark2SqlParser.scala]: Combine both showCache and showexecutor Cache parsers

like (SHOW ~> opt(EXECUTOR) <~ METACACHE) ~ opt(ontable)

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/1583/

File: integration/spark2/src/main/scala/org/apache/carbondata/indexserver/DistributedShowCacheRDD.scala (hunk 1 above)
Comment (vikramahuja1001) [reply integration/spark2/src/main/scala/org/apache/carbondata/indexserver/DistributedShowCacheRDD.scala]: done

File: integration/spark2/src/main/scala/org/apache/carbondata/indexserver/IndexServer.scala (hunk 2 above)
Comment (vikramahuja1001) [reply integration/spark2/src/main/scala/org/apache/carbondata/indexserver/IndexServer.scala]: done

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/1584/

File: integration/spark2/src/main/scala/org/apache/carbondata/indexserver/IndexServer.scala (hunk 3 above)
Comment (vikramahuja1001) [reply integration/spark2/src/main/scala/org/apache/carbondata/indexserver/IndexServer.scala]: done

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala (hunk 4 above)
Comment (vikramahuja1001) [reply integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala]: done

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala (hunk 5 above)
Comment (vikramahuja1001) [reply integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala]: done

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala (hunk 6 above)
Comment (vikramahuja1001) [reply integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala]: done

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala (hunk 7 above)
Comment (vikramahuja1001) [reply integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala]: done

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala (hunk 8 above)
Comment (vikramahuja1001) [reply integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala]: done

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala (hunk 9 above)
Comment (vikramahuja1001) [reply integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala]: done

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala (hunk 10 above)
Comment (vikramahuja1001) [reply integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala]: done

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala (hunk 11 above)
Comment (vikramahuja1001) [reply integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala]: done

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala (hunk 12 above)
Comment (vikramahuja1001) [reply integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala]: done

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala (hunk 13 above)
Comment (vikramahuja1001) [reply integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala]: done

File: integration/spark2/src/main/scala/org/apache/spark/sql/parser/CarbonSpark2SqlParser.scala (hunk 14 above)
Comment (vikramahuja1001) [reply integration/spark2/src/main/scala/org/apache/spark/sql/parser/CarbonSpark2SqlParser.scala]: done

**Comment by vikramahuja1001** [comment]
retest this please

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/1587/

**Comment by vikramahuja1001** [comment]
retest this please

**Comment by vikramahuja1001** [comment]
retest this please

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/1590/

**Comment by vikramahuja1001** [comment]
retest this please

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/1615/

File: integration/spark-carbon-common-test/src/test/scala/org/apache/carbondata/sql/commands/TestCarbonShowCacheCommand.scala [hunk 15]
```diff
@@ -175,13 +175,13 @@ class TestCarbonShowCacheCommand extends QueryTest with BeforeAndAfterAll {
     // Empty database
     sql("use cache_empty_db").collect()
     val result1 = sql("show metacache").collect()
-    assertResult(2)(result1.length)
-    assertResult(Row("cache_empty_db", "ALL", "0 B", "0 B", "0 B", "DRIVER"))(result1(1))
+    sql("show metacache").show()
```
Comment (kunal642) [inline integration/spark-carbon-common-test/src/test/scala/org/apache/carbondata/sql/commands/TestCarbonShowCacheCommand.scala]: Remove this

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala [hunk 16]
```diff
@@ -87,7 +108,8 @@ case class CarbonShowCacheCommand(tableIdentifier: Option[TableIdentifier],
       val indexRawResults = if (CarbonProperties.getInstance().isDistributedPruningEnabled
       (tableIdentifier.get.database.getOrElse(sparkSession.catalog.currentDatabase),
         tableIdentifier.get.table)) {
-        getTableCacheFromIndexServer(carbonTable, numberOfIndexFiles)(sparkSession)
+        getTableCacheFromIndexServer(carbonTable,
+          numberOfIndexFiles)(showExecutorCache)(sparkSession)
```
Comment (kunal642) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala]: Revert this change

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala [hunk 17]
```diff
@@ -110,13 +132,41 @@ case class CarbonShowCacheCommand(tableIdentifier: Option[TableIdentifier],
...
   }
 
+  def getAllExecutorCache(sparkSession: SparkSession): Seq[Row] = {
+    val isDistributedPruningEnabled = CarbonProperties.getInstance()
+      .isDistributedPruningEnabled("", "")
+    if (!isDistributedPruningEnabled) {
+      // Block here. this feature is only with index server enabled
+      throw new UnsupportedOperationException(
+        "Show Executor Metacache is only avalable with Index Server Enabled")
+      Seq()
```
Comment (kunal642) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala]: no need to return Seq() if you are already throwing exception

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala [hunk 18]
```diff
@@ -169,41 +224,32 @@ case class CarbonShowCacheCommand(tableIdentifier: Option[TableIdentifier],
...
-          "INDEX SERVER")
+        Row("TOTAL", indexAllIndexSize, indexAllDatamapSize, indexAllDictSize, "INDEX SERVER")
       ) ++ indexServerRows).collect {
-        case row if row.getLong(2) != 0L || row.getLong(3) != 0L || row.getLong(4) != 0L =>
-          Row(row.get(0), row.get(1), bytesToDisplaySize(row.getLong(2)),
-            bytesToDisplaySize(row.getLong(3)), bytesToDisplaySize(row.getLong(4)), "INDEX SERVER")
+        case row if row.getLong(1) != 0L || row.getLong(2) != 0L || row.getLong(3) != 0L =>
+          Row(row.get(0),
+            bytesToDisplaySize(row.getLong(1)),
+            bytesToDisplaySize(row.getLong(2)),
```
Comment (kunal642) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala]: revert this

File: integration/spark2/src/main/scala/org/apache/spark/sql/parser/CarbonSpark2SqlParser.scala [hunk 19]
```diff
@@ -560,9 +560,9 @@ class CarbonSpark2SqlParser extends CarbonDDLSqlParser {
     }
 
   protected lazy val showCache: Parser[LogicalPlan] =
-    SHOW ~> METACACHE ~> opt(ontable) <~ opt(";") ^^ {
-      case table =>
-        CarbonShowCacheCommand(table)
+    (SHOW ~> opt(EXECUTOR) <~ METACACHE) ~ opt(ontable) <~ opt(";") ^^ {
```
Comment (kunal642) [inline integration/spark2/src/main/scala/org/apache/spark/sql/parser/CarbonSpark2SqlParser.scala]: add this DDL change in documentation

**Comment by vikramahuja1001** [comment]
retest this please

File: integration/spark-carbon-common-test/src/test/scala/org/apache/carbondata/sql/commands/TestCarbonShowCacheCommand.scala (hunk 15 above)
Comment (vikramahuja1001) [reply integration/spark-carbon-common-test/src/test/scala/org/apache/carbondata/sql/commands/TestCarbonShowCacheCommand.scala]: done

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala (hunk 18 above)
Comment (vikramahuja1001) [reply integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala]: done

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/1653/

**Comment by vikramahuja1001** [comment]
retest this please

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/1655/

**Comment by vikramahuja1001** [comment]
retest this please

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/1667/

**Comment by vikramahuja1001** [comment]
@kunal642 , done

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/1677/

**Comment by kunal642** [comment]
LGTM

**Comment by kunal642** [comment]
@vikramahuja1001 please rebase

**Comment by vikramahuja1001** [comment]
@kunal642 , done

**Comment by vikramahuja1001** [comment]
retest this please

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/1726/
